sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -1520,6 +1521,292 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
             }
         }
         #endregion GetBlockList
+
+        #region SetExpiry
+        /// <summary>
+        /// Sets the blob's ExpiresOn property.  ExpiresOn is when
+        /// the blob will be deleted.  If <paramref name="relativeToBlobCreationTime"/>
+        /// is true, the ExpiresOn time will be set relative to the
+        /// time the blob was created.  Otherwise, it will be set
+        /// relative to the current time.
+        /// </summary>
+        /// <param name="timeToExpire">
+        /// <see cref="TimeSpan"/>.
+        /// </param>
+        /// <param name="relativeToBlobCreationTime">
+        /// If true, the ExpiresOn property will be set relative to
+        /// the time the blob was created.  Otherwise, it will be set
+        /// relative to the current time.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobInfo}"/> describing the blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<BlobInfo> SetExpiryRelative(
+            TimeSpan timeToExpire,
+            bool relativeToBlobCreationTime = default,
```

Comment:
I'd make this enum rather than boolean. Easier for somebody reading code that uses it to understand what it does without checking what's the name of bool flag. Besides if another point to be relative to is added then it will be easier to extend.

Reply:
Changed to an enum for both Blobs and Data Lake.